gateway/reverse_proxy.go

```diff
@@ -1845,3 +1848,12 @@ func (p *ReverseProxy) IsUpgrade(req *http.Request) (string, bool) {
 
 	return httputil.IsUpgrade(req)
 }
+
+func (p *ReverseProxy) addAuthInfo(outReq, req *http.Request) {
+	if !p.TykAPISpec.UpstreamAuth.IsEnabled() {
+		return
+	}
+
+	authProvider := ctxutil.GetUpstreamAuth(req)
```

Comment:
This needs a conditional, what if authProvider == nil (no upstream auth enabled)?